src/main/java/uk/gov/hmcts/ccd/domain/types/CaseDataValidator.java

```diff
@@ -124,10 +129,19 @@ public CaseDataValidator(final List<FieldValidator> validators) {
                     new RuntimeException("System error: No validator found for " + fieldType.getType()));
     }
 
-    private boolean isPredefinedTypeFieldValidator(FieldValidator validator, String fieldID) {
-        if (validator instanceof PredefinedTypeFieldValidator) {
-            String predefinedId = ((PredefinedTypeFieldValidator) validator).getPredefinedFieldId();
-            return predefinedId.equals(fieldID);
+    private boolean isCustomTypeValidator(FieldValidator validator, String fieldTypeId) {
+        if (validator instanceof CustomTypeValidator) {
+            String customTypeId = ((CustomTypeValidator) validator).getCustomTypeId();
+            return customTypeId.equals(fieldTypeId);
+        }
+        return false;
+    }
+
+    private boolean isFieldIdBasedValidator(FieldValidator validator, String fieldId) {
+        if (validator instanceof FieldIdBasedValidator) {
+            ((FieldIdBasedValidator) validator).setValidationContext(validationContext);
```

Comment:
is it possible to remove this and pass the validation context around as a parameter to the validate method?

Reply:
DONE